docs/book/src/cronjob-tutorial/testdata/project/controllers/cronjob_controller_test.go

```diff
@@ -0,0 +1,200 @@
+Without the stubbed Job template spec and the Pod template spec below, the Kubernetes API will not be able to
+create the CronJob.
+*/
+var _ = Describe("CronJob Controller", func() {
+
+	// Define utility constants for object names and testing timeouts and intervals.
+	const (
+		CronjobName      = "test-cronjob"
+		CronjobNamespace = "test-cronjob-namespace"
+		JobName          = "test-job"
+
+		timeout  = time.Second * 10
+		interval = time.Millisecond * 250
+	)
+
+	Context("Update CronJob Status", func() {
+		It("CronJob Status.Active count should increase when active Jobs are created", func() {
+			ctx := context.Background()
+
+			cronJob := &cronjobv1.CronJob{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "batch.tutorial.kubebuilder.io/v1",
+					Kind:       "CronJob",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      CronjobName,
+					Namespace: CronjobNamespace,
+				},
+				Spec: cronjobv1.CronJobSpec{
+					Schedule: "1 * * * *",
+					JobTemplate: batchv1beta1.JobTemplateSpec{
+						Spec: batchv1.JobSpec{
+							// For simplicity, we only fill out the required fields.
+							Template: v1.PodTemplateSpec{
+								Spec: v1.PodSpec{
+									// For simplicity, we only fill out the required fields.
+									Containers: []v1.Container{
+										{
+											Name:  "test-container",
+											Image: "test-image",
+										},
+									},
+									RestartPolicy: v1.RestartPolicyOnFailure,
+								},
+							},
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, cronJob)).Should(Succeed())
+
+			/*
+				After creating this CronJob, let's check that the CronJob's Spec fields match what we passed in.
+				Note that, because the k8s apiserver may not have finished creating a CronJob after our `Create()` call from earlier, we will use Gomega’s Eventually() testing function instead of Expect() to give the apiserver an opportunity to finish creating our CronJob.
+
+				`Eventually()` will repeatedly run the function provided as an argument every interval seconds until
+				(a) the function’s output matches what’s expected in the subsequent `Should()` call, or
+				(b) the number of attempts * interval period exceed the provided timeout value.
+
+				In the examples below, timeout and interval are Go Duration values of our choosing.
+			*/
+			By("CronJob spec should match arguments passed in during creation")
+			cronjobLookupKey := types.NamespacedName{Name: CronjobName, Namespace: CronjobNamespace}
+			createdCronjob := &cronjobv1.CronJob{}
+
+			// We'll need to retry getting this newly created CronJob, given that creation may not immediately happen.
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, cronjobLookupKey, createdCronjob)
+				if err != nil {
+					return false
+				}
+				return true
+			}, timeout, interval).Should(BeTrue())
+			Expect(createdCronjob.Spec.Schedule).Should(Equal("1 * * * *"))
```

Comment:
can you add a comment here or something explaining this?  I'm assuming it's a sanity test for when you have pruning turned on on the API server (CRD v1 or CRD v1beta1 w/ conversion), but it's worth pointing out.

Reply:
This isn't as much a sanity test for pruning as it is a check to ensure that our CronJob spec `Schedule` field matches the parameters we passed in. (I don't think `Schedule` is supposed to ever get pruned?) Please let me know if I misunderstood something!

Reply:
I think i'm just not understanding the failure condition you're testing for.  The two failure modes I can imaging are conversion screwing up something during round-tripping (which I forgot about when leaving the comment here) or pruning removing the field.  Otherwise, "field passed in" == "field read" should be ensured by the apiserver.

Reply:
This failure condition was to check for "field passed in" == "field read" (which is ensured by the APIserver). The idea for adding this came from some conversations @camilamacedo86 and I had earlier (see https://github.com/kubernetes-sigs/kubebuilder/pull/1521#discussion_r431373610). Given that the consistency of a created object and its input spec is ensured by/tested in the apiserver, I think it honestly makes sense to just delete this check--what do you think?

Reply:
I went with the second option (add extra comment to clarify that this tests the conversion of the Schedule string).